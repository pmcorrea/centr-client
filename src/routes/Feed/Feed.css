.App__main > .feed_ul {
  padding: 0px;
  margin: 0 auto;
}

.feed_li {
  color: black;
}

.post_box {
  color: black;
  background-color: rgba(238, 238, 238, 0.63);
  border: none;
  border-radius: 10px 10px 0px 0px;
  margin: 20px auto;
  padding: 10px 20px;
  border-bottom: 2px solid purple;
  box-shadow: 0px 5px 5px rgba(128, 128, 128, 0.733);
  max-width: 300px;
}

.post_title,
.post_title > a {
  font-size: 20px;
  color: rgb(112, 39, 161);
  margin: 0px;
  margin-top: 15px;
  text-decoration: none;
  text-transform: capitalize;
}

.post_title,
.post_title > a:hover {
  color: orange;
}

.post_user {
  color: black;
  font-size: 16px;
  margin: 0px;
  font-weight: 500;
}

.post_modified {
  color: gray;
  font-size: 12px;
  margin: 0px;
  font-weight: 500;
}

.post_content {
  color: black;
  font-weight: 400;
  font-size: 14px;
  margin-top: 0px;
  margin-bottom: 0px;
  line-height: 1.4;
}

.no_friends_img {
  width: 200px;
  background-color: transparent;
  margin: 0 auto;
  display: block;
  margin-top: 100px;
}

.no_friends_caption {
  margin: 0 auto;
  padding: 20px;
  color: black;
  display: block;
  text-align: center;
  font-size: 30px;
  width: 300px;
}

.post_subbox {
  display: flex;
  flex-direction: row;
  justify-content: left;
  align-items: center;
  margin-bottom: 15px;
}

.post_subbox_2 {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 10px;
}

.feed_avatar {
  width: 40px;
  height: 40px;
  border-radius: 50%;
}

.section_headers {
  margin: 0 auto;
  font-size: 30px;
  font-weight: 500;
  color: black;
  font-family: "Pacifico", cursive;
  text-align: center;
  display: block;
}

.icons {
  width: 60px;
  height: 60px;
  margin: 0 auto;
  padding: 0px;
  display: block;
  top: 0px;
  box-shadow: none;
}

label,
select {
  display: block;
  width: 100%;
  margin: 0 auto;
}

/* 786 and smaller */
@media only screen and (max-width: 768px) {
  .icons_bar {
    width: 80vw;
    background-color: rgb(112, 39, 161);
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    z-index: 100;
    position: fixed;
    bottom: 0px;
    padding: 10px;
    height: 60px;
    border-radius: 15px 15px 0px 0px;
    left: 50%;
    margin-left: -40vw;
  }
}

/* 7869and larger */
@media only screen and (min-width: 769px) {
  .feed_ul {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    width: 80vw;
  }

  .feed_li {
    list-style: none;
  }

  .request_ul,
  .following_ul,
  .discover_ul,
  .FolderView__List {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
  }

  .post_box {
    color: black;
    background-color: rgba(238, 238, 238, 0.63);
    border: none;
    border-radius: 10px 10px 0px 0px;
    margin: 20px;
    padding: 10px 20px;
    border-bottom: 2px solid purple;
    box-shadow: 0px 5px 5px rgba(128, 128, 128, 0.733);
    min-width: 200px;
    max-width: 300px;
  }

  .discover_li {
    width: auto;
  }

  .FolderView__List {
    margin: 0 auto;
  }

  .request_li,
  .following_li {
    width: 30vw;
    height: 10vh;
    list-style: none;
    margin-bottom: 5vh;
  }

  .FoldersView {
    width: 400px;
    margin: 0 auto;
  }

  .profile_box {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 350px;
  }

  .AccountPage__forms {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .LoginPage {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  /* .LoginPage> input, .LoginPage> button {
		width: 200px;
	} */

  /* .RegisterForm input, .RegisterForm>button {
		width: 200px;
		display: block;
		margin: 0 auto;
		margin-bottom: 10px;
	} */

  .registration__validation_box {
    width: 300px;
    display: block;
    margin: 0 auto;
    margin-bottom: 10px;
  }

  /* .AddPost__form>input, .AddPost__form>select, .AddPost__form>button, .AddPost__form>textarea, .AddPost__form>label {
		display: block;
		width: 300px;
		margin: 0 auto;
	} */

  .AddPost__form > select,
  .AddPost__form > textarea,
  .AddPost__form > label {
    display: block;
    width: 300px;
    margin: 0 auto;
  }

  .AddPost__form {
    margin-top: 10vh;
  }

  .AddPost__form > label {
    margin-top: 20px;
  }

  .AddPost__form > textarea {
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* .AddFolder__form>input, .AddFolder__form>select, .AddFolder__form>button, .AddFolder__form>textarea, .AddFolder__form>label {
		display: block;
		width: 300px;
		margin: 0 auto;	
		margin-bottom: 20px;	
	} */

  .AddFolder__form > select,
  .AddFolder__form > textarea,
  .AddFolder__form > label {
    display: block;
    width: 300px;
    margin: 0 auto;
    margin-bottom: 20px;
  }

  .PublicPostRoute {
    width: 60vw;
    margin: 0 auto;
    margin-top: 5vh;
  }

  .PostsList {
    width: 600px;
    margin: 0 auto;
  }

  .Post {
    width: auto;
  }

  .PostWithContent {
    width: 600px;
    margin: 0 auto;
  }

  .EditPost__form_container {
    width: 600px;
    margin: 0 auto;
  }

  .EditPost__form {
    width: auto;
  }

  .icons_bar {
    width: 80vw;
    min-width: 300px;
    margin: 0 auto;
    background-color: rgb(112, 39, 161);
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    z-index: 100;
    bottom: 0px;
    padding: 10px;
    height: 60px;
    border-radius: 15px 15px 0px 0px;
    position: fixed;
    left: 50%;
    margin-left: -40vw;
  }
}
