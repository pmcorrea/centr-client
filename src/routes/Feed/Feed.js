import React, { Component } from "react";
import "./Feed.css";
import AuthApiService from "../../services/auth-api-service";
import MainContext from '../../contexts/MainContext'
import { Link } from "react-router-dom";


export default class Feed extends Component {

	state = {
		connections: [],
		error: null,
		requests: [],
		currentUser: null,
		publicPosts: [],
		showLoader: false
	}

  	static contextType = MainContext;

	componentDidMount() {
		this.setCurrentUser()
		this.setConnections()
		setTimeout(() => {
			if(this.state.publicPosts.length === 0){
				this.setState({ showLoader: true })
			}
		}, 500);
	}

	setCurrentUser() {
		AuthApiService.getCurrentUserDetailsById()
		.then(currentUser => {
			currentUser = currentUser[0]

			this.setState({
				currentUser: currentUser
			})
		})
		.catch(error => {
			this.setState({ error }, () => console.log(this.state.error))
		})
	}

	// Grab connections and set posts
	setConnections() {
		AuthApiService.getConnections()
		.then(connections => {
			if (connections) {
				this.setState({
					connections: connections
				}, () => this.setPublicPosts())
			}
		})
		.catch(error => {
			this.setState({ error }, () => console.log(this.state.error))
		})
	}

	setPublicPosts() {
		let connections = this.state.connections
		let connectionsIds = connections.map(x => {
		return x.id
		})
    
		AuthApiService.getPublicPosts(connectionsIds)
			.then(result => {

				result.forEach(x => {
					let newDate = new Date(x.modified)
					let timeNow = new Date() 
					let diff = timeNow.getTime() - newDate.getTime()
					diff = diff/(1000*60*60*24)
					diff = diff.toFixed(0)
					x.modified = diff
				})

				this.setState({
					publicPosts: result
				})
			})
			.catch(error => {
					this.setState({ error }, () => console.log(this.state.error))
			})
  	}
  
	randomKey() {
		return Math.random().toString(36).substr(2, 9);
	}

  render() {
	return this.state.requests 
	? (<>
		<p className="section_headers">Feed</p>    
		<ul className="feed_ul">
			{(this.state.publicPosts.length !== 0) 
				? this.state.publicPosts.map(post => (
					<li className="feed_li" key={`${post['post_name']}${this.randomKey()}`}>
						<div className="post_box">

							<div className="post_subbox">
								<img src={post.avatar} alt="avatar" className="feed_avatar"></img>
								
								<div className="post_subbox_2">
									<p className="post_user">{post['user_name']}</p>
									<p className="post_modified">{`${post['modified']} days ago`}</p>
								</div>	
							</div>

							<h2 className="post_title">
								<Link to={`/PublicPost/${post['id']}`}>{post['post_name']}</Link>
							</h2>
							<p className="post_content">{`${post['content'].substring(0, 150)}...`}</p>
						</div>
					</li>)) 
				: ('')
			}

			{(this.state.showLoader) 
				? <div>
					<img src="../../../assets/no_friends_unicorn.png" alt="sad unicorn with no friends" className="no_friends_img"/>
					<p className="no_friends_caption">You have no friends...go get some!</p>
				</div>
				: ('')
			}
		</ul>
	</>) 
	: ('')
  }
}