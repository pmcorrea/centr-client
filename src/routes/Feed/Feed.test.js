import Feed from './Feed.js'
import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('App test suite', () => {
	// Smoke Test
	it('renders without crashing', () => {
		const div = document.createElement('div')

		ReactDOM.render(
			<Feed />,
			div
		)
		ReactDOM.unmountComponentAtNode(div)
	})

	// Snapshot Test
	it('renders the snapshot', () => {
		const wrapper = shallow(<Feed />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})