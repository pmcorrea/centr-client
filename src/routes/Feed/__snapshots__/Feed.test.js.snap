// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite renders the snapshot 1`] = `
<Fragment>
  <p
    className="section_headers"
  >
    Feed
  </p>
  <ul
    className="feed_ul"
  />
</Fragment>
`;
